core/commands/partials/vagrant_configuration_generator.rb

```diff
@@ -222,14 +223,19 @@ def make_node_params(node, box_params)
     if node[1].key?('box_parameters')
       symbolic_box_params = override_box_params(node, symbolic_box_params)
     end
-    {
-      name: node[0].to_s,
-      host: node[1]['hostname'].to_s,
-      vm_mem: node[1]['memory_size'].nil? ? '1024' : node[1]['memory_size'].to_s,
-      vm_cpu: (@env.cpu_count || node[1]['cpu_count'] || '1').to_s,
-      private_ip: node[1]['private_ip'].nil? ? nil : node[1]['private_ip'].to_s,
-      default_route: node[1]['default_route'].nil? ? nil : node[1]['default_route'].to_s
-    }.merge(symbolic_box_params)
+
+    if (!node[1]['private_ip'].nil? && !node[1]['default_route'].nil?) || (node[1]['private_ip'].nil? && node[1]['default_route'].nil?)
```

Comment:
This should be moved into a function with a clear name. This boolean logic seems too complex.